src/khepri_machine.erl

```diff
@@ -1253,6 +1253,12 @@ handle_aux(
                Projection <- Projections]
       end),
     {no_reply, AuxState, LogState};
+handle_aux(
+  _RaState, cast, {update_projections, OldState}, AuxState, LogState,
+  NewState) ->
+    ok = update_projections(OldState, NewState),
```

Comment:
could this work not be done in `snapshot_installed/4` instead of delegating to aux?

Reply:
yeah good point, I've removed the `update_projections` effect and moved the handler for that inline here